### ISSUE TITLE
SystemVideoView doesn't show controls

### ISSUE THREAD

**Issue by hoschiho** [SRGSSR/pillarbox-apple#806]
### Description of the problem

Using the SystemVideoView instead of VideoView, doesn't display the iOS specific default controls.

I was able to display them, by adding `controller.beginAppearanceTransition(true, animated: false)` in BasicSystemVideoView.swift:

```swift

import AVKit
import SwiftUI
import PillarboxPlayer

struct MyBasicSystemVideoView: UIViewControllerRepresentable {
    let player: Player
    let gravity: AVLayerVideoGravity

#if os(tvOS)
    func makeCoordinator() -> AVPlayerViewControllerSpeedCoordinator {
        .init()
    }
#endif

    func makeUIViewController(context: Context) -> AVPlayerViewController {
        let controller = AVPlayerViewController()
        controller.beginAppearanceTransition(true, animated: false)
        controller.allowsPictureInPicturePlayback = false
#if os(tvOS)
        controller.playbackControlsIncludeInfoViews = true
#endif
        return controller
    }

    func updateUIViewController(_ uiViewController: AVPlayerViewController, context: Context) {
        uiViewController.player = player.systemPlayer
        uiViewController.videoGravity = gravity
#if os(tvOS)
        context.coordinator.player = player
        context.coordinator.controller = uiViewController
#endif
    }
}
``` 
source: https://forums.developer.apple.com/forums/thread/711360?wt_debug=1&wtstp_debug=1

### Reproducibility

Always

### Steps to reproduce

Using SystemVideoView instead of VideoView

### Library version

1.0.1

### Operating system

ios 17

### Is there an existing issue for this?

- [X] I have searched existing issues and found no similar bug report.

**Comment by hoschiho** [comment]
Hello @defagos 
Yes, the system player controls are never displayed and can't be toggled. What's interesting, once I use the SystemVideoView, I can play & pause the video by clicking in the middle (where usually the play button would be), without any visual feedback. 

We're using a UIHostingController to display the SwiftUI view:

```
import Foundation
import Flutter
import UIKit
import SwiftUI
import PillarboxPlayer

class FLNativeView: NSObject, FlutterPlatformView {
    private var _view: UIView

    init(
        frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?,
        binaryMessenger messenger: FlutterBinaryMessenger?,
        playerWrapper: PlayerWrapper
    ) {
        _view = UIView()
        super.init()
        let childView = UIHostingController(rootView: VideoPlayerView(playerWrapper: playerWrapper))
        childView.view.frame = frame
        _view.addConstrained(subview: childView.view)
        childView.didMove(toParent:_view.inputViewController)


    }

    func view() -> UIView {
        return _view
    }
}

extension UIView {
    func addConstrained(subview: UIView) {
        addSubview(subview)
        subview.translatesAutoresizingMaskIntoConstraints = false
        subview.topAnchor.constraint(equalTo: topAnchor).isActive = true
        subview.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
        subview.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
        subview.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
    }
}
```


And the VideoPlayerView is just the SystemVideoView:

```
var body: some View {
        SystemVideoView(player: playerWrapper.player)
    }
```


I hope this helps.

**Comment by defagos** [comment]
Thanks for the information, we'll have a look with `UIHostingController` involved.

**Comment by defagos** [comment]
I implemented the following scenarios with a custom `PlayerView` embedding `SystemVideoView` and an associated player:

- Display a `UIHostingController` hosting this view as root view controller.
- Display a `UIHostingController` hosting this view, itself embedded as child of another view controller displayed as root view controller.
- Push a `UIHostingController` hosting this view into a navigation controller.
- Display a `UIHostingController` hosting this view, itself embedded as child of another view controller pushed into a navigation controller.
- Present a `UIHostingController` hosting this view modally.
- Display a `UIHostingController` hosting this view, itself embedded as child of another view controller presented modally.

In all these use cases the behavior is fine.

I then applied the fix suggested in the [link](https://forums.developer.apple.com/forums/thread/711360?wt_debug=1&wtstp_debug=1) you provide. This breaks Picture in Picture automatic restoration on tvOS, though, in the following case:

- Play some content.
- Enter PiP.
- Click on the same content again. Restoration fails to merge back the PiP layer into the originating view.

We'll investigate further but it might be helpful if we can build, run and debug your project. This way we can avoid introducing a fix that may do more harm than good.

**Comment by defagos** [comment]
Looking at your code it feels strange that `inputViewController` is involved as parent controller since this controller is related to keyboard input (active responder). See [associated documentation](https://developer.apple.com/documentation/uikit/uiresponder/1621117-inputviewcontroller).

Parent-child containment relationships are very important for correct view lifecycle propagation. Maybe this is what explains the issue your observe, or maybe Flutter does not maintain correct containment relationships somewjere.

If you can provide us with either access to your code, or to a small example reproducing this issue in Flutter (we don't know how you integrate Pilllarbox so we are a bit in the dark), we would be able to better investigate view and controller hierarchies to discover what might be misssing, either in your implementation or in Flutter.

If you prefer we can also attempt to debug your code together.

Thanks in advance for your help.

**Comment by waliid** [comment]
Hi @hoschiho,

First, thank you for providing us access to your code.

So, after some investigations, we have found a solution to use directly the Pillarbox `SystemVideoView` instead of reinventing the wheel with an `UIViewControllerRepresentable` and adding a specific behavior such as calling `beginAppearanceTransition(_:animated:)`.
Hence, instead of using `MyBasicSystemVideoView` we are now able to use `SystemVideoView` in the `VideoPlayerView`.

I've tried to improve some part of code related to the playback, and I also allowed myself to create a [PR](https://github.com/bitforge/rtr/pull/77) with some improvements in my point of view.
